### ISSUE TITLE
Make FiberRuntime settings configurable

### ISSUE THREAD

**Issue by ak0rz** [zio/zio#7542]
see https://github.com/zio/zio/blob/series/2.x/core/shared/src/main/scala/zio/internal/FiberRuntime.scala#L1474-L1478

```scala
  private[zio] final val MaxTrampolinesBeforeYield = 5
  private[zio] final val MaxOperationsBeforeYield  = 1024 * 10
  private[zio] final val MaxDepthBeforeTrampoline  = 300
  private[zio] final val MaxWorkStealingDepth      = 150
  private[zio] final val WorkStealingSafetyMargin  = 50
```

If we make these configurable, it may allow end-users to make advanced fine-tuning of runtime to adapt it to their workloads.

In order to help this tuning, we may add metrics like number of limit-yields and limit-trampolines over the time.

**Comment by adamgfraser** [comment]
Not sure it makes sense to expose these as opposed to runtime flags like enabling work stealing as they are extremely low level and might change in future optimizations to the runtime but it is certainly possible.

**Comment by ak0rz** [comment]
> Not sure it makes sense to expose these as opposed to runtime flags like enabling work stealing as they are extremely low level and might change in future optimizations to the runtime but it is certainly possible.

Of course, it's should be marked as unstable API and most of the users wouldn't change them in any way.

But it's like java runtime and GC, you know. Better have some knobs that you'll never use than not to have them. :)

**Comment by jdegoes** [comment]
These settings can have a big impact on the runtime but like @adamgfraser says, we can't guarantee they exist from one release to the next, or even that they have the same meaning if they do continue to exist. It's possible we could support their configuration via system properties or environment variables.
